test/controllers/registrations_controller_test.rb

```diff
@@ -55,12 +55,55 @@ class RegistrationsControllerTest < ActionDispatch::IntegrationTest
       ]
     )
 
-    post(
-      callback_registration_url,
-      params: { credential_nickname: "USB Key" }.merge(public_key_credential)
-    )
+    assert_no_difference -> { User.count } do
+      post(
+        callback_registration_url,
+        params: { credential_nickname: "USB Key" }.merge(public_key_credential)
```

Comment:
What do you think about also adding a test where the `crendential.verify` method fails and ensure that the user is not created?

Reply:
I've covered it in my test, I just wasn't at ease doing it in Minitest.

```ruby
describe "POST #callback" do
  context "with invalid params" do
    let(:fake_rp) do
      instance_double(WebAuthn::FakeRelyingParty)
    end

    it "renders a response with 422 status" do
      allow(fake_rp).to receive(:verify_registration).and_raise(WebAuthn::Error, "WebAuthn::UserVerifiedVerificationError")
      allow(controller).to receive(:relying_party).and_return(fake_rp)

      session = { current_registration: { challenge: "Challenge" } }

      expect do
        expect do
          post :callback, params: { registration: invalid_attributes }, session:
        end.not_to change(User, :count)
      end.not_to change(Credential, :count)

      expect(response).to have_http_status(:unprocessable_entity)

      expect(response.parsed_body).to eq({ "error" => "Verification failed: WebAuthn::UserVerifiedVerificationError" })
    end
  end
end
```

Reply:
Cool, then don't worry about it. I'll see what I can do 🙂